rust/share/agama.libsonnet

```diff
@@ -0,0 +1,26 @@
+// function go throught lshw output and enlist only given class.
+// Basically it is same as calling `lshw -class <class>`.
+selectClass(lshw, class)::
+  local selectClass_(parent, class) =
+    if std.objectHas(parent, 'class') && parent.class == class then
+      [ parent ]
+    else if std.objectHas(parent, 'children') then
+      std.flattenArrays(std.prune(std.map(function(x) selectClass_(x, class), parent.children )))
+    else
+      [];
+
+  local result = selectClass_(lshw, class);
+  result,
+
+// function go throught lshw output and returns object with given "id" or null if not found.
+findID(lshw, id)::
```

Comment:
About the name of the function, you are not "finding" and ID, but an object. So I would rename it to `findById` or something similar (or `findObject` or whatever).

Reply:
Please, document the arguments (type and so on).